docs: add basic README for each Helm chart on their own directory

fmuyassarov opened the pull request:
This commit adds basic documentation for each Helm chart so that basic information is available to the user who are trying to install any of those plugins via artifacthub.io registry.
An example: https://artifacthub.io/packages/helm/node-feature-discovery/node-feature-discovery

klihub commented:
You need to link to those new bits of documentation from somewhere which is reachable from the top entry point. Otherwise the doc build check/verification refuses to accept it.

fmuyassarov commented:
> You need to link to those new bits of documentation from somewhere which is reachable from the top entry point. Otherwise the doc build check/verification refuses to accept it.

Fixed and moved out some bits from installation.md. PTAL.

fmuyassarov commented:
failed again...I will check

klihub commented on deployment/helm/balloons/README.md:
```diff
@@ -0,0 +1,44 @@
+
+| Name                     | Default                                                                                                                       | Description                                          |
+| ------------------------ | ----------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------- |
+| `image.name`             | [ghcr.io/containers/nri-plugins/nri-resource-policy-balloons](ghcr.io/containers/nri-plugins/nri-resource-policy-balloons)    | container image name                                 |
```
The link should be an URL, like `[ghcr.io/containers/...](https://ghcr.io/containers/...)`

klihub commented on deployment/helm/balloons/README.md:
```diff
@@ -0,0 +1,44 @@
+| `hostPort`               | 8891                                                                                                                          | metrics port to expose on the host                   |
+| `config`                 | <pre><code>ReservedResources:</code><br><code>  cpu: 750m</code></pre>                                                        | plugin configuration data                            |
+| `nri.patchRuntimeConfig` | false                                                                                                                         | enable NRI in containerd or CRI-O                    |
+| `initImage.name`         | [ghcr.io/containers/nri-plugins/config-manager](ghcr.io/containers/nri-plugins/config-manager)                                | init container image name                            |
```
The link should be an URL, like `[ghcr.io/containers/...](https://ghcr.io/containers/...)`

klihub commented on deployment/helm/topology-aware/README.md:
```diff
@@ -0,0 +1,45 @@
+
+| Name                     | Default                                                                                                                       | Description                                          |
+| ------------------------ | ----------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------- |
+| `image.name`             | [ghcr.io/containers/nri-plugins/nri-resource-policy-topology-aware](ghcr.io/containers/nri-plugins/nri-resource-policy-topology-aware)    | container image name                     |
```
The link should be an URL, like `[ghcr.io/containers/...](https://ghcr.io/containers/...)`

klihub commented on deployment/helm/topology-aware/README.md:
```diff
@@ -0,0 +1,45 @@
+| `hostPort`               | 8891                                                                                                                          | metrics port to expose on the host                   |
+| `config`                 | <pre><code>ReservedResources:</code><br><code>  cpu: 750m</code></pre>                                                        | plugin configuration data                            |
+| `nri.patchRuntimeConfig` | false                                                                                                                         | enable NRI in containerd or CRI-O                    |
+| `initImage.name`         | [ghcr.io/containers/nri-plugins/config-manager](ghcr.io/containers/nri-plugins/config-manager)                                | init container image name                            |
```
The link should be an URL, like `[ghcr.io/containers/...](https://ghcr.io/containers/...)`

klihub commented on deployment/helm/memtierd/README.md:
```diff
@@ -0,0 +1,43 @@
+
+| Name                     | Default                                                                                                                       | Description                                          |
+| ------------------------ | ----------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------- |
+| `image.name`             | [ghcr.io/containers/nri-plugins/nri-memtierd](ghcr.io/containers/nri-plugins/nri-memtierd)                                    | container image name                                 |
```
The link should be an URL, like `[ghcr.io/containers/...](https://ghcr.io/containers/...)`

klihub commented on deployment/helm/memtierd/README.md:
```diff
@@ -0,0 +1,43 @@
+| `resources.memory`       | 100Mi                                                                                                                         | memory qouta for the                                 |
+| `outputDir`              | empty string                                                                                                                  | host directory for memtierd.output files             |
+| `nri.patchRuntimeConfig` | false                                                                                                                         | enable NRI in containerd or CRI-O                    |
+| `initImage.name`         | [ghcr.io/containers/nri-plugins/config-manager](ghcr.io/containers/nri-plugins/config-manager)                                | init container image name                            |
```
The link should be an URL, like `[ghcr.io/containers/...](https://ghcr.io/containers/...)`

fmuyassarov commented:
> @fmuyassarov I played a bit around with how we could include the helm chart docs among our generated documentation while keeping them in their current (very logical and natural) location in deployment/helm/*/README.md. I ended up having a helm directory under docs with its own ToC which links to each helm chart doc (I had to slightly adjust the Helm chart docs to make the end result look more sensible). The actual helm chart docs are stitched into the the main doc tree using includes. And the main doc includes the helm ToC, like we do for the rest of the documentation.
> 
> Here is the whole shebang, with two additional commits on top of your tree: https://github.com/klihub/nri-plugins/tree/docs/add-helm-readme

Thanks @klihub. I think those changes you have made in the other two commits are good to have. Because first they fix the issue with the doc building and second reorganize the docs to have a separate section describing each chart parameters and installation/un-installation procedure. I have now cherry-picked your commits. PTAL.
/cc @kad

fmuyassarov commented:
Actually I might need to rename the commit [docs: experiment with Helm chart doc inclusion.](https://github.com/containers/nri-plugins/pull/167/commits/91854dc41f60561af0feb198c2620e21d8c3e2e1)

klihub commented:
> Actually I might need to rename the commit [docs: experiment with Helm chart doc inclusion.](https://github.com/containers/nri-plugins/pull/167/commits/91854dc41f60561af0feb198c2620e21d8c3e2e1)

Sure, that should be reworded.

klihub commented:
@fmuyassarov I would also squash at least the second fixup+headings commit to the first one. There is no benefit in keeping it separate.

marquiz reviewed (changes requested):
Hmm, `make site-build` fails with 

```
...
/bin/bash: -c: line 1: unexpected EOF while looking for matching `''
/bin/bash: -c: line 3: syntax error: unexpected end of file
make: *** [Makefile:412: html] Error 2
make: *** [Makefile:426: site-build] Error 2
```

klihub commented:
@fmuyassarov Can you rebase this on latest main/HEAD ? Then @marquiz should be able to `make site-[build|serve]` from a git worktree.

klihub reviewed (changes requested):
Otherwise LGTM, but could you rebase this on top of latest main/HEAD. Helps review{ers,ing} locally...

marquiz commented on docs/helm/index.md:
```diff
@@ -0,0 +1,12 @@
+# Helm Charts
```
Should we actually rename this to more generic `Deployment` (and the rename the subdir too) instead? Would probably be easier for people to navigate the documentation.

```suggestion
# Deployment
```

Then in this file shortly mention that currently only Helm charts are supported. We can add kustomize later if we're confident about that :)  WDYT?

fmuyassarov replied:
+1. Fixed.

marquiz reviewed (commented):
Thanks @fmuyassarov for the update. I think this makes sense and is definitely a good improvement 👍 One comment below about naming.

Also, I think you should remove all the helm stuff from `docs/resource-policy/installation.md`. Just add a link to the new location or smth?

fmuyassarov commented:
> Also, I think you should remove all the helm stuff from `docs/resource-policy/installation.md`. Just add a link to the new location or smth?

@marquiz Yes, I think that's a good point. Because all the prerequisites would also appear in the plugin's corresponding README.md.
I have now update the PR. PTAL. And, you should be able to run `make site-serve `.

marquiz reviewed (approved):
LGTM
